### PR TITLE
1.15.0rc1 feedstock

### PR DESCRIPTION
**Pull request by johnkerl**

### CODE REVIEW

File: conda-forge.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ build_platform:
 conda_forge_output_validation: true
 test: native_and_emulated
 # build_with_mambabuild: false
-upload_on_branch: main
+upload_on_branch: 1.15.0rc1
```
Comment (jdblischak) [inline conda-forge.yml:10]: For this to work, the target branch for merging needs to be `1.15.0rc1`. You'll need to create it manually first:

```sh
git checkout main
git checkout -b 1.15.0rc1
# assuming you name it origin
git push origin 1.15.0rc1
```

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 2]
```diff
@@ -35,7 +31,7 @@ jobs:
         export sha=$(Build.SourceVersion)
         export GIT_BRANCH=$BUILD_SOURCEBRANCHNAME
         export FEEDSTOCK_NAME=$(basename ${BUILD_REPOSITORY_NAME})
-        export UPLOAD_ON_BRANCH="main"
+        export UPLOAD_ON_BRANCH="1.15.0rc0"
```
Comment (jdblischak) [inline .azure-pipelines/azure-pipelines-linux.yml:34]: You need to rerender the feedstock. You changed `upload_on_main` to `1.15.0rc1`, but the CI files are still using `1.15.0rc0`

**Review by jdblischak — CHANGES_REQUESTED** [review]
Two things:

* Need to coordinate the upload branch. Currently `conda-forge.yml`, the Azure CI files, and the PR target branch are all different
* Less critical, but wouldn't be a bad idea to make explicit the new requirement for `setuptools >=70.1`

**Review by jdblischak — APPROVED** [review]

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 2 above)
Comment (johnkerl) [reply .azure-pipelines/azure-pipelines-linux.yml]: pushed ...

**Review by ihnorton — APPROVED** [review]

**Pull request merged**